rasa/core/training/story_writer/yaml_story_writer.py

```diff
@@ -0,0 +1,146 @@
+from collections import OrderedDict
+
+import ruamel.yaml as ruamel_yaml
+from typing import List, Text, Union, Optional
+
+from rasa.utils.common import raise_warning
+from ruamel.yaml.scalarstring import DoubleQuotedScalarString
+
+from rasa.constants import LATEST_TRAINING_DATA_FORMAT_VERSION, DOCS_URL_STORIES
+from rasa.core.events import UserUttered, ActionExecuted, SlotSet, Form
+from rasa.core.training.story_reader.yaml_story_reader import (
+    KEY_STORIES,
+    KEY_STORY_NAME,
+    KEY_USER_INTENT,
+    KEY_ENTITIES,
+    KEY_ACTION,
+    KEY_STEPS,
+    KEY_CHECKPOINT,
+    KEY_SLOT_NAME,
+    KEY_SLOT_VALUE,
+    KEY_CHECKPOINT_SLOTS,
+    KEY_OR,
+)
+from rasa.core.training.structures import StoryStep, Checkpoint
+
+import rasa.utils.io as io_utils
+
+
+class YAMLStoryWriter:
+    def dumps(self, story_steps: List[StoryStep]) -> Text:
+        """Turns Story steps into a string."""
+        stream = ruamel_yaml.StringIO()
+        self.dump(stream, story_steps)
+        return stream.getvalue()
+
+    def dump(
+        self, target: Union[Text, ruamel_yaml.StringIO], story_steps: List[StoryStep],
+    ) -> None:
+        from rasa.validator import KEY_TRAINING_DATA_FORMAT_VERSION
+
+        self.target = target
+
+        stories = []
+        for story_step in story_steps:
+            processed_story_step = self.process_story_step(story_step)
+            if processed_story_step:
+                stories.append(processed_story_step)
+
+        result = OrderedDict()
+        result[KEY_TRAINING_DATA_FORMAT_VERSION] = DoubleQuotedScalarString(
+            LATEST_TRAINING_DATA_FORMAT_VERSION
+        )
+        result[KEY_STORIES] = stories
+
+        io_utils.write_yaml(result, self.target, True)
+
+    def process_story_step(self, story_step: StoryStep) -> Optional[OrderedDict]:
+
+        if self.story_contains_forms(story_step):
+            raise_warning(
+                f'File "{self.target}" contains a story "{story_step.block_name}" '
+                f"that has form(s) in it. This story cannot be converted automatically "
+                f"because of the new Rules system in the Rasa Open Source "
+                f"{LATEST_TRAINING_DATA_FORMAT_VERSION} version."
```

Comment:
```suggestion
                f"because of the new Rules system in Rasa Open Source "
                f"version {LATEST_TRAINING_DATA_FORMAT_VERSION}."
```